src/dp_abstraction/aca_net_state_handler.cpp

```diff
@@ -0,0 +1,215 @@
+// Copyright 2019 The Alcor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "aca_log.h"
+#include "aca_dataplane_mizar.h"
+#include "aca_net_state_handler.h"
+#include "goalstateprovisioner.grpc.pb.h"
+#include <future>
+
+using namespace alcorcontroller;
+
+std::mutex gs_reply_mutex; // mutex for writing gs reply object
+
+namespace aca_net_state_handler
+{
+Aca_Net_State_Handler::Aca_Net_State_Handler()
+{
+  ACA_LOG_INFO("Network State Handler initialize\n");
+
+  // default to dataplane_mizar for now
+  ACA_LOG_INFO("Network State Handler: using mizar dataplane\n");
+  this->core_net_programming_if = new aca_dataplane_mizar::ACA_Dataplane_Mizar;
+}
+
+Aca_Net_State_Handler::~Aca_Net_State_Handler()
+{
+  // allocated core_net_programming_if is destroyed when program exits.
+}
+
+Aca_Net_State_Handler &Aca_Net_State_Handler::get_instance()
+{
+  // It is instantiated on first use.
+  // allocated instance is destroyed when program exits.
+  static Aca_Net_State_Handler instance;
+  return instance;
+}
+
+int Aca_Net_State_Handler::update_vpc_state_workitem(const VpcState current_VpcState,
+                                                     GoalStateOperationReply &gsOperationReply)
+{
+  return this->core_net_programming_if->update_vpc_state_workitem(
+          current_VpcState, std::ref(gsOperationReply));
+}
+
+int Aca_Net_State_Handler::update_vpc_states(GoalState &parsed_struct,
+                                             GoalStateOperationReply &gsOperationReply)
+{
+  std::vector<std::future<int> > workitem_future;
```

Comment:
workitem_queue?

Reply:
the name is to align with the c++ "future" implementation of the threading model used in the code.

Reply:
Cool thanks.